src/css.js

```diff
@@ -257,7 +264,11 @@ jQuery.extend( {
 			if ( !hooks || !( "set" in hooks ) ||
 				( value = hooks.set( elem, value, extra ) ) !== undefined ) {
 
-				style[ name ] = value;
+				if ( isCustomProp ) {
```

Comment:
I'm fine with breaking that out for a later fix, it does seem like all of this can be refactored better at that time.